mitiq/_about.py

```diff
@@ -15,55 +15,155 @@
 
 """Information about Mitiq and dependencies."""
 import platform
+from typing import Dict
+import warnings
+from pkg_resources import parse_requirements
 
 from cirq import __version__ as cirq_version
 from numpy import __version__ as numpy_version
 from scipy import __version__ as scipy_version
 
 import mitiq
 
+# Logging versions of Mitiq's core dependencies (must be installed)
+LATEST_SUPPORTED_PKGS = {
+    req.project_name: req.specs[0][1]
+    for req in parse_requirements(open("requirements.txt"))
+}
+INSTALLED_PKGS = {}
+INSTALLED_PKGS["cirq"] = cirq_version
+INSTALLED_PKGS["numpy"] = numpy_version
+INSTALLED_PKGS["scipy"] = scipy_version
+
+# Checking versions of Mitiq's optional dependencies (may be installed)
+OPTIONAL_PKGS = [
+    "pyquil",
+    "qiskit",
+    "amazon-braket-sdk",
+    "pennylane",
+    "pennylane-qiskit",
+]
+
+try:
+    from pyquil import __version__ as pyquil_version
+
+    INSTALLED_PKGS["pyquil"] = pyquil_version
+except ImportError:
+    INSTALLED_PKGS["pyquil"] = "Not installed"
+try:
+    from qiskit import __qiskit_version__  # pragma: no cover
+
+    INSTALLED_PKGS["qiskit"] = __qiskit_version__["qiskit"]  # pragma: no cover
+except ImportError:
+    INSTALLED_PKGS["qiskit"] = "Not installed"
+try:
+    from braket._sdk import __version__ as braket_version
+
+    INSTALLED_PKGS["amazon-braket-sdk"] = braket_version
+except ImportError:
+    INSTALLED_PKGS["amazon-braket-sdk"] = "Not installed"
+try:
+    from pennylane import __version__ as pennylane_version
+
+    INSTALLED_PKGS["pennylane"] = pennylane_version
+except ImportError:
+    INSTALLED_PKGS["pennylane"] = "Not installed"
+try:
+    from pennylane_qiskit import __version__ as pennylane_qiskit_version
+
+    INSTALLED_PKGS["pennylane-qiskit"] = pennylane_qiskit_version
+except ImportError:
+    INSTALLED_PKGS["pennylane-qiskit"] = "Not installed"
+
+
+def latest_optional_supported() -> Dict:
+    """Returns the versions of Mitiq's optional packages that are supported by
+    the current version of Mitiq. Requires that the dependency has a pinned
+    version in the dev_requirements.txt file.
+    """
+    return {
+        req.project_name: req.specs[0][1]
+        for req in parse_requirements(open("dev_requirements.txt"))
+        if req.project_name in (OPTIONAL_PKGS)
+    }
+
+
+LATEST_SUPPORTED_PKGS.update(latest_optional_supported())
+
+
+def latest_supported_packages() -> Dict:
+    """Returns the versions of Mitiq's optional packages that are supported by
+    the current version of Mitiq. Requires that the dependency has a pinned
+    version in the dev_requirements.txt file.
+    """
+    return LATEST_SUPPORTED_PKGS
+
+
+def check_versions() -> None:
+    """Checks that the installed versions of Mitiq's dependencies are
+    supported by the current version of Mitiq.
+    """
+
+    for name, version in INSTALLED_PKGS.items():
+        if LATEST_SUPPORTED_PKGS[name] != version:
```

Comment:
Maybe this is a too strict condition which risks to create too many warnings. 
What about if the warning is produced only if `LATEST_SUPPORTED_PKGS[name] < version` ?

Note: if we go for this weaker check, [this tool](https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python) may be necessary to compare versions.